test/runtime/symlink.c

```diff
@@ -0,0 +1,121 @@
+#include <errno.h>
+#define _GNU_SOURCE
+#define __USE_GNU
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#define test_assert(expr) do { \
+    if (!(expr)) { \
+        printf("Error: %s -- failed at %s:%d\n", #expr, __FILE__, __LINE__); \
+        exit(EXIT_FAILURE); \
+    } \
+} while (0)
+
+int main(int argc, char **argv)
+{
+    int fd;
+    char buf[8];
+    struct stat s;
+    char *cwd;
+
```

Comment:
May want to add setbuf(stdout, NULL) so we don't lose any error output on a crash.